src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+package com.minecolonies.rankup.util;
+
+import com.minecolonies.rankup.Rankup;
+import com.minecolonies.rankup.modules.core.config.AccountConfigData;
+import com.minecolonies.rankup.modules.databases.DatabaseModule;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.service.sql.SqlService;
+import org.spongepowered.api.service.user.UserStorageService;
+import uk.co.drnaylor.quickstart.exceptions.NoModuleException;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * Lots of accounts based utils (Used for database / file config adaption)
+ */
+public class AccountingUtils extends ConfigUtils
+{
+
+    private SqlService sql;
+    private Connection conn;
+
+    private              String table_id           = "player_stats";
+    private static final String UUID_COLUMN        = "UUID";
+    private static final String PLAYER_NAME_COLUMN = "PLAYER_NAME";
+    private static final String JOIN_DATE_COLUMN   = "JOIN_DATE";
+    private static final String LAST_JOIN_COLUMN   = "LAST_JOIN";
+    private static final String TIME_PLAYED_COLUMN = "TIME_PLAYED";
+
+    public AccountingUtils(final Rankup pl)
+    {
+        super(pl);
+    }
+
+    private DataSource getDataSource(String jdbcUrl) throws SQLException
+    {
+        table_id = getDatabasesConfig().sqlTablePrefix + "player_stats";
+        if (sql == null)
+        {
+            sql = Sponge.getServiceManager().provide(SqlService.class).get();
+        }
+        return sql.getDataSource(jdbcUrl);
+    }
+
+    private String getURI()
+    {
+        if (getDatabasesConfig().database.equalsIgnoreCase("mysql"))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "mysql" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1132)